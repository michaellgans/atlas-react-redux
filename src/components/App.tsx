function App() {
  return <>Hello Atlas</>;
}

export default App;
