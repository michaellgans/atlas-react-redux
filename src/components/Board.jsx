// Board Component

// Asset Imports

// Script Imports

// Returns a Board Component
export function Board() {
    // Define Hook
    return (
    );
}
