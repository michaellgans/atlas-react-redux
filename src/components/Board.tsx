// Board Component

// Asset Imports
import { useAppSelector } from "../store";
import { List } from "./List";
import { RootState } from "../store";

// Returns a Board Component
export function Board() {
  // Define Hook
  const lists = useAppSelector((state: RootState) => state.listsSlice.items);

  return (
    // START BOARD
    <div className="m-auto h-screen w-screen overflow-x-scroll text-center">
      <div className="flex h-full space-x-4">
        {lists.map((list) => (
          <List key={list.id} listID={list.id} listTitle={list.title} />
        ))}
      </div>
    </div>
  );
}
