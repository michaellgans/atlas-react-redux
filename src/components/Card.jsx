// Card Component

// Asset Imports

// Script Imports

// Returns a Card Component
export function Card() {
    // Define Hook
    return (
    );
}
