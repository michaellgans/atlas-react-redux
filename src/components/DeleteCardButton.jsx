// DeleteCardButton Component

// Asset Imports

// Script Imports

// Returns a DeleteCardButton Component
export function DeleteCardButton() {
    // Define Hook
    return (
    );
}
