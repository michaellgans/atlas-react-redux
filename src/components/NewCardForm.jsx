// NewCardForm Component

// Asset Imports

// Script Imports

// Returns a NewCardForm Component
export function NewCardForm() {
    // Define Hook
    return (
    );
}
