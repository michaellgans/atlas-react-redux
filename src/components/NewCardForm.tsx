// NewCardForm Component

// Asset Imports

// Returns a NewCardForm Component
export function NewCardForm() {
  // Define Hook
  return (
    // START NEW CARD FORM
    <div className="group/new-card m-3 flex h-44 w-full justify-center">
      {/* onSubmit="alert('Create card')" */}
      <form className="hidden min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 text-blue group-hover/new-card:flex">
        <input
          className="w-11/12 resize-none overflow-auto rounded-t-3xl border-0 bg-off-white-light px-0 py-6 text-xl font-black text-blue outline-none"
          autoFocus
          type="text"
          placeholder="Title"
          name="title"
        />
        <textarea
          className="w-11/12 resize-none overflow-auto border-0 bg-off-white-light text-blue outline-none"
          placeholder="Description"
          name="description"
        ></textarea>
        <div className="buttons w-full">
          <button type="submit" className="w-full p-4">
            Save
          </button>
        </div>
      </form>
    </div>
    // END NEW CARD FORM
  );
}
