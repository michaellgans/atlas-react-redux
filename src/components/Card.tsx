// Card Component

// Asset Imports
import { DeleteCardButton } from "./DeleteCardButton";

// Script Imports

// Types
type CardProps = {
  title: string;
  description: string;
};

// Returns a Card Component
export function Card({ title, description }: CardProps) {
  // Define Hook
  return (
    <div className="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue">
      <h5 className="my-2 flex w-full items-end justify-between text-xl font-black">
        <span>{title}</span>
        {/* DELETE CARD BUTTON */}
        <DeleteCardButton />
      </h5>
      <p className="mt-0 text-left">{description}</p>
    </div>
  );
}
