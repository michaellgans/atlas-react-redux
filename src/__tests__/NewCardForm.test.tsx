// Test for NewCardForm Component

// Imports
import { render } from "@testing-library/react";
import { expect, test, it } from 'vitest';
import { NewCardForm } from "../components/NewCardForm";

it('Controll Test', () => {
    expect(2 + 2).toBe(4);
});

// test('NewCardForm renders correctly', () => {
//     const { container } = render(<NewCardForm />);
//     expect(container).toMatchSnapshot();
// });
