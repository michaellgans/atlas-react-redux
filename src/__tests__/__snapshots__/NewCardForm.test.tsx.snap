// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewCardForm renders correctly 1`] = `
<div>
  <div
    class="group/new-card m-3 flex h-44 w-full justify-center"
  >
    <form
      class="hidden min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 text-blue group-hover/new-card:flex"
    >
      <input
        class="w-11/12 resize-none overflow-auto rounded-t-3xl border-0 bg-off-white-light px-0 py-6 text-xl font-black text-blue outline-none"
        name="title"
        placeholder="Title"
        type="text"
      />
      <textarea
        class="w-11/12 resize-none overflow-auto border-0 bg-off-white-light text-blue outline-none"
        name="description"
        placeholder="Description"
      />
      <div
        class="buttons w-full"
      >
        <button
          class="w-full p-4"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
